rotkehlchen/tests/unit/accounting/test_basic.py

```diff
@@ -44,7 +44,23 @@ def test_simple_accounting(accountant, google_service, expected_pnls):
 
 
 @pytest.mark.parametrize('mocked_price_queries', [prices])
-def test_selling_crypto_bought_with_crypto(accountant, google_service):
+@pytest.mark.parametrize(('db_settings', 'expected_pnl_totals'), [
+    (
+        {'include_fees_in_cost_basis': True},
+        PnlTotals({
+            AccountingEventType.TRADE: PNL(taxable=FVal('40.58809373210132103047999998'), free=ZERO),  # noqa: E501
```

Comment:
Here the difference is because I changed the prices calculation to be more consistent.

So previously:

1. Here https://github.com/rotki/rotki/blob/bugfixes/rotkehlchen/accounting/pot.py#L384 `out_price` was preferred
2. But here https://github.com/rotki/rotki/blob/bugfixes/rotkehlchen/accounting/pot.py#L392 `in_price` was preferred

Now what happened is that:
1. I changed the calculation so that `in_price` is checked first everywhere
2. In this test `rate` here https://github.com/rotki/rotki/blob/bugfixes/rotkehlchen/tests/unit/accounting/test_basic.py#L56 is quite different from the mocked price here https://github.com/rotki/rotki/blob/f87130989c512d60b05aabde96889bdc22dccd75/rotkehlchen/tests/utils/history.py#L72. So when I adjusted the calculation logic, the difference ended up being quite large.

Reply:
Hmmm what do you mean by "is preferred"? I do not follow this explanation. The links you gave me to bugfixes before compute `calculated_in` depending on existence and value of `out_price`, and `calculated_out` depending on existence and value of `in_price`.

Reply:
Discussed and settled on the following priority:

- If asset_out or asset_in is fiat --> use price of the fiat asset.
- If neither of the assets is fiat --> use out_price if known, otherwise in_price.